packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart

```diff
@@ -276,6 +278,99 @@ class CpuProfileData {
 
   factory CpuProfileData.empty() => CpuProfileData.parse({});
 
+  /// Generates [CpuProfileData] by fetching [CpuSamples] from the [VmService]
+  /// that is stored in the global [ServiceConnectionManager].
+  ///
+  /// Only samples collected in the time range [origin, origin + extent] will be reported.
+  ///
+  /// [isolateId] The id which is passed to the getIsolate RPC to load this isolate.
+  /// [origin] origin time
+  /// [extent] extent time
+  static Future<CpuProfileData> generateFromCpuSamples(
+    String isolateId,
+    int origin,
+    int extent,
+  ) async {
+    // As of service protocol version 3.27 _getCpuProfileTimeline does not exist
+    // and has been replaced by getCpuSamples. We need to do some processing to
+    // get back to the format we expect.
+    final cpuSamples =
+        await serviceManager.service!.getCpuSamples(isolateId, origin, extent);
```

Comment:
push this call into `getCpuProfile` in `cpu_profile_service.dart`. Then make this method accept just two params `String isolateId` and `CpuSamples cpuSamples`.